pkg/object/object.go

```diff
@@ -236,9 +295,85 @@ func (v *Volume) validate() error {
 	return nil
 }
 
+func (sd *SecretData) validate() error {
+	var count int
+	if sd.Plaintext != nil {
+		count++
+	}
+	if sd.Base64 != nil {
+		count++
+	}
+	if sd.File != nil {
+		count++
+	}
+
+	switch count {
+	case 0:
+		return fmt.Errorf("Please set one of plaintext, base64 or file field for the secret key: %v", sd.Key)
+	case 1:
+		return nil
+	case 2, 3:
+		return fmt.Errorf("Only one of plaintext, base64 or file fields can be set at a time for the secret key: %v", sd.Key)
+	default:
+		return fmt.Errorf("Something went wrong with counting the secret fields for the secret key: %v", sd.Key)
+	}
+}
+
+func validateSecretRef(sRef *string, oSec *[]Secret) error {
+	// validate secretRef syntax
+	secretDef, err := util.SecretRefToSecretDef(sRef)
+	if err != nil {
+		return fmt.Errorf("unable to verify secretRef syntax: %v", err)
+	}
+
+	if len(*oSec) == 0 {
+		log.Printf("There are no root level secrets defined, assuming the corresponding secret exists in the cluster: %v", *sRef)
+	} else {
+		secretFound := false
+		for _, secret := range *oSec {
+			if secretDef.SecretName == secret.Name {
+				secretFound = true
+
+				dataKeyFound := false
+				for _, secData := range secret.Data {
```

Comment:
implement a search function on the secret root level object similar to one here in [`VolumeExists`](https://github.com/redhat-developer/opencompose/blob/master/pkg/object/object.go#L92)